src/org/labkey/test/util/AuditLogHelper.java

```diff
@@ -69,6 +80,117 @@ public DataRegionTable goToAuditEventView(String eventType)
+
+            // filter out SampleStateLabel as that is not a change, it is added for display purposes
+            dataChanges = Stream.of(dataChanges).filter(s -> !s.toLowerCase().startsWith("samplestatelabel=")).toArray(String[]::new);
+            // filter out RowId as that is not a change, it is added for display purposes
+            dataChanges = Stream.of(dataChanges).filter(s -> !s.toLowerCase().startsWith("rowid=")).toArray(String[]::new);
+
+            TestLogger.log("Audit record data changes diff count check: " + dataChangesStr);
+//            _wrapper.checker().verifyEquals("Audit record data changes did not include the expected number of diffs, expected " + expectedDiffCount + " but was " + dataChanges.length + ": " + dataChangesStr,
+//                    expectedDiffCount, dataChanges.length);
+            assertEquals("Audit record data changes did not include the expected number of diffs, expected " + expectedDiffCount + " but was " + dataChanges.length + ": " + dataChangesStr,
+                    expectedDiffCount, dataChanges.length);
+        }
+    }
+
+    /**
+     * Check for th expected number of diffs in the audit event for the last transactionId.
+     * If an expectedEventCount is also provided, it will check that the number of events for that transactionId matches the expectedEventCount.
+     */
+    public void checkTimelineAuditEventDiffCountForLastTransaction(String containerPath, String auditEventName, int expectedDiffCount, @Nullable Integer expectedEventCount) throws IOException, CommandException
+    {
+        Integer transactionId = (Integer) getAuditLogsFromLKS(containerPath, auditEventName, List.of("TransactionId"), Collections.emptyList(), 1)
+                .getRows().get(0).get("TransactionId");
+        List<Filter> transactionFilter = List.of(new Filter("TransactionId", transactionId, Filter.Operator.EQUAL));
+        int eventCount = getAuditLogsFromLKS(containerPath, auditEventName, List.of("NewRecordMap"), transactionFilter, null).getRows().size();
+        if (expectedEventCount != null)
+//            _wrapper.checker().verifyEquals("Unexpected number of events for transactionId " + transactionId, expectedEventCount.intValue(), eventCount);
+            assertEquals("Unexpected number of events for transactionId " + transactionId, expectedEventCount.intValue(), eventCount);
+        List<Integer> expectedChangeCounts = Collections.nCopies(eventCount, expectedDiffCount);
+        checkTimelineAuditEventDiffCount(containerPath, auditEventName, expectedChangeCounts);
+    }
+
+    public static String getAuditEventNameFromURL()
+    {
+        if (isSamplesRoute())
+            return "SampleTimelineEvent";
+        else if (isDataClassRoute())
+            return "SourcesAuditEvent";
+        return null;
+    }
+
+    public static boolean isSamplesRoute()
+    {
+        URL url = BaseWebDriverTest.getCurrentTest().getURL();
+        if (url != null)
```

Comment:
Doe these route checking methods need to be static? Want to avoid using `BaseWebDriverTest.getCurrentTest()`.
```suggestion
    public boolean isSamplesRoute()
    {
        URL url = _wrapper.getURL();
        if (url != null)
```

Reply:
No need to be static anymore. Changed.